src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file InverseKinematic.h
+ * @authors Davide Gorbani <davide.gorbani@iit.it>
+ */
+
+#ifndef BIOMECHANICAL_ANALYSIS_INVERSE_KINEMATIC_H
+#define BIOMECHANICAL_ANALYSIS_INVERSE_KINEMATIC_H
+
+// iDynTree
+#include <iDynTree/KinDynComputations.h>
+
+// BipedalLocomotion
+#include <BipedalLocomotion/IK/QPInverseKinematics.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/VariablesHandler.h>
+#include <BipedalLocomotion/IK/SO3Task.h>
+#include <BipedalLocomotion/ContinuousDynamicalSystem/FloatingBaseSystemKinematics.h>
+#include <BipedalLocomotion/ContinuousDynamicalSystem/ForwardEuler.h>
+
+namespace BiomechanicalAnalysis
+{
+
+namespace IK
+{
+
+class HumanIK
+{
+private:
+    // Integration time step in nanoseconds
+    std::chrono::nanoseconds m_dtIntegration;
+
+    // Struct to integrate the base and joint velocities
+    struct System
+    {
+        std::shared_ptr<BipedalLocomotion::ContinuousDynamicalSystem::ForwardEuler<BipedalLocomotion::ContinuousDynamicalSystem::FloatingBaseSystemKinematics>> integrator;
+        std::shared_ptr<BipedalLocomotion::ContinuousDynamicalSystem::FloatingBaseSystemKinematics> dynamics;
+    };
+
+    // System to integrate the base and joint velocities
+    System m_system;    
+
+    // Joint positions and velocities
+    Eigen::VectorXd m_jointPositions;
+    Eigen::VectorXd m_jointVelocities;
+    Eigen::Matrix4d m_basePose;
+    Eigen::Matrix<double, 6, 1> m_baseVelocity;
+    Eigen::Vector3d m_gravity;
+
+    // tasks
+    std::shared_ptr<BipedalLocomotion::IK::SO3Task> m_link1OrientationTask;
+
+    // pointer to the KinDynComputations object
+    std::shared_ptr<iDynTree::KinDynComputations> m_kinDyn;
+
+    // Number of Joint Degrees of Freedom
+    int m_nrDoFs;
+
+    BipedalLocomotion::IK::QPInverseKinematics m_qpIK;
+    BipedalLocomotion::System::VariablesHandler m_variableHandler;
+
+public:
+    HumanIK(){}; // constructor
+    ~HumanIK(){}; // destructor
```

Comment:
I would document the code using doxygen format. Similar to the one in bipedal locomotion framework. This will can be used to automatically generate the documentation

Reply:
Ok, I will do it!

Reply:
I was thinking something more structured. Like https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/src/IK/include/BipedalLocomotion/IK/QPFixedBaseInverseKinematics.h